moor_generator/lib/src/model/view.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:moor_generator/src/analyzer/options.dart';
+import 'package:moor_generator/src/analyzer/runner/file_graph.dart';
+import 'package:sqlparser/sqlparser.dart';
+
+import 'base_entity.dart';
+import 'declarations/declaration.dart';
+import 'model.dart';
+
+/// A parsed view
+class MoorView extends MoorSchemaEntity {
+  @override
+  final MoorViewDeclaration declaration;
+
+  /// The associated view to use for the sqlparser package when analyzing
+  /// sql queries. Note that this field is set lazily.
+  View parserView;
+
+  final String name;
+
+  @override
+  List<MoorSchemaEntity> references = [];
+
+  List<ViewColumn> columns;
```

Comment:
This is probably fine, but we could consider wrapping these with a `MoorColumn`, similar to what `MoorTable` is doing.

Reply:
We don't need the `columns` right now so I removed it, later we can add it back if we need it.

`MoorColumn` is defined for a table so it has parameters such as `customConstraints` that are not needed for a View. Maybe we can create a custom Column class for views but until then I'll remove it in my next commit.